src/QsCompiler/CompilationManager/ProjectManager.cs

```diff
@@ -793,11 +826,12 @@ public bool ManagerTask(Uri file, Action<CompilationUnitManager> executeTask, ID
             Action<Exception>? exceptionLogger,
             Action<string, MessageType>? log = null,
             Action<PublishDiagnosticParams>? publishDiagnostics = null,
-            SendTelemetryHandler? sendTelemetry = null)
+            SendTelemetryHandler? sendTelemetry = null,
+            BuildConfiguration? buildConfiguration = null)
         {
             this.load = new ProcessingQueue(exceptionLogger);
             this.projects = new ConcurrentDictionary<Uri, Project>();
-            this.defaultManager = new CompilationUnitManager(ProjectProperties.Empty, exceptionLogger, log, publishDiagnostics, syntaxCheckOnly: true);
+            this.defaultManager = new CompilationUnitManager(ProjectProperties.Empty, exceptionLogger, log, publishDiagnostics, syntaxCheckOnly: true, buildConfiguration);
```

Comment:
Carrying this comment over too: https://github.com/microsoft/qsharp-compiler/pull/1457#discussion_r907862698

I'm not sure about it, though, since `BuildConfiguration` is maintained by the `EditorState` and passed to the `ProjectManager`. So would the `ProjectManager` create a new `BuildConfiguration` with `syntaxCheckOnly=true`? It's possible but I am struggling to see the benefit

Reply:
Is this comment moot now? It doesn't look like this line is changed in the current version anymore.

Reply:
Yeah, when changing isNotebook to be per-file, I sent `BuildConfiguration` to the farm upstate